### PR TITLE
[docs] Add contributor agreement prereq to dev docs

### PR DESCRIPTION
**Pull request by mdavis332**
## What does this PR do?

CI builds fail if the [elastic contributor agreement](https://www.elastic.co/contributor-agreement) isn't on file for a github user. However, I didn't see that info anywhere prior to trying to submit my first integration so my initial CI build failed due to it. 

This PR adds that language to the dev docs about building a new integration, specifically in the **Open a PR** section.

## Related issues

- See #3839 for a PR that failed due to no contributor agreement on file at time PR was submitted.

### CODE REVIEW

**Comment by mdavis332** [comment]
i signed the contributor agreement a few days ago shortly after my CI check for #3839 failed. just a heads up for any elastic team member looking at this, perhaps there's some manual work needed once the agreement is submitted before it registers with the CI pipeline? i.e., I would have expected the CLA check to _not_ fail today since I signed it several days ago.

**Comment by jsoriano** [comment]
Hey @mdavis332, looking to the information about you in the CLA checker, it looks like you signed the CLA with a different email address than the one used in this commit. You need to sign the CLA with the same email address that you use in your commits.

You can find here the raw patch for the commit of this PR, where your email appears: https://github.com/elastic/integrations/commit/73fcc63fe3c9a7ef5f233833368df7d7ab6f7a78.patch

Could you try to modify your commit to use the same email address you used to sign the CLA? Or sign the CLA with the email you are using in your commits?

**Comment by mdavis332** [comment]
Hi @jsoriano, thank you for the help pointing that out! Git still routinely confounds me and I (mistakenly) intuited that the cla-checker-service would use the github username supplied in the contributor agreement rather than the `user.email` in the git config. 

As my git-foo is such that it would take me longer to figure out how to fix that commit, I just re-signed the agreement with the email addr set in that config. Will that work?

**Comment by jsoriano** [comment]
> Will that work?

CLA checker seems happy now, so it seems it worked :) thanks!

File: docs/developer_workflow_design_build_test_integration.md [hunk 1]
```diff
@@ -119,7 +119,9 @@ using the tool's [documentation](https://github.com/elastic/elastic-package/tree
 
 ### Open a PR
 
-If you think that you've finished works on your integration, you've verified that it collects data, wrote some tests,
+Prior to opening a PR, you must sign the [elastic contributor agreement](https://www.elastic.co/contributor-agreement) if you haven't already. CI builds will fail if this agreement isn't on file for your GitHub username.
```
Comment (jsoriano) [inline docs/developer_workflow_design_build_test_integration.md:122]: For the CI to run, an approval from someone from Elastic is needed.
```suggestion
Prior to opening a PR, you must sign the [elastic contributor agreement](https://www.elastic.co/contributor-agreement) if you haven't already.
```

Comment (mdavis332) [reply docs/developer_workflow_design_build_test_integration.md:122]: @jsoriano just making sure you aren't waiting on me for anything here.

Comment (jsoriano) [reply docs/developer_workflow_design_build_test_integration.md:122]: Yes, sorry, I was suggesting to remove the last sentence in this line, let me know what you think.

CLA agreement is not enough to pass CI, an approval from someone from Elastic is also needed.

Comment (mdavis332) [reply docs/developer_workflow_design_build_test_integration.md]: ok, sounds good to me. i mostly just want to ensure that external contributors have docs on everything that they can/should do before issuing a PR. i think this addresses the missing CLA component. thank you!

**Review by jsoriano — APPROVED** [review]

**Pull request merged**